lib/utils/fs_test.go

```diff
@@ -26,6 +26,80 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
+func TestOpenFileSymlinks(t *testing.T) {
+	// symlink structure setup, this will produce the following structure under the temp directory created below:
+	// dir
+	// dir-s -> dir
+	// dir-s-s -> dir-s
+	// dir/file
+	// dir/file-s -> dir/file
+	// circular-s -> circular-s
+	// broken-s -> nonexistent
+	rootDir, err := os.MkdirTemp("", "symlink-test")
+	require.NoError(t, err)
+	defer os.RemoveAll(rootDir)
+
+	dirPath := filepath.Join(rootDir, "dir")
+	err = os.Mkdir(dirPath, 0755)
+	require.NoError(t, err)
+
+	filePath := filepath.Join(dirPath, "file")
+	_, err = os.Create(filePath)
+	require.NoError(t, err)
+
+	symlinkToFile := filepath.Join(dirPath, "file-s")
+	err = os.Symlink(filePath, symlinkToFile)
+	require.NoError(t, err)
+
+	symlinkDir := filepath.Join(rootDir, "dir-s")
+	err = os.Symlink(dirPath, symlinkDir)
+	require.NoError(t, err)
+
+	symlinkToSymlinkDir := filepath.Join(rootDir, "dir-s-s")
+	err = os.Symlink(symlinkDir, symlinkToSymlinkDir)
+	require.NoError(t, err)
+
+	circularSymlink := filepath.Join(rootDir, "circular-s")
+	err = os.Symlink(circularSymlink, circularSymlink)
+	require.NoError(t, err)
+
+	brokenSymlink := filepath.Join(rootDir, "broken-s")
+	err = os.Symlink(filepath.Join(rootDir, "nonexistent"), brokenSymlink)
+	require.NoError(t, err)
+
+	// Tests below can not be done with t.Parallel due to the need for directory cleanup in defer
+	t.Run("directFileOpenAllowed", func(t *testing.T) {
+		f, err := openFile(filePath, false)
+		require.NoError(t, err)
+		defer f.Close()
+	})
```

Comment:
This one actually fails on my machine, because `/var` is symlink to `/private/var`:

```
=== RUN   TestOpenFileSymlinks/directFileOpenAllowed
    fs_test.go:73: 
        	Error Trace:	/Users/tener/code/teleport/lib/utils/fs_test.go:73
        	Error:      	Received unexpected error:
        	            	symlink not allowed in path: /var
        	Test:       	TestOpenFileSymlinks/directFileOpenAllowed
    --- FAIL: TestOpenFileSymlinks/directFileOpenAllowed (0.00s)
```

I wonder if we can make the error more informative, i.e. include the information about the file we were trying to access:
```
=== RUN   TestOpenFileSymlinks/directFileOpenAllowed
    fs_test.go:77: 
        	Error Trace:	/Users/tener/code/teleport/lib/utils/fs_test.go:77
        	Error:      	Received unexpected error:
        	            	failed to open file /var/folders/c0/_6x4ddps6zn5js_4y31xlz380000gn/T/TestOpenFileSymlinks2354108266/001/dir/file, symlink not allowed in path: /var
        	Test:       	TestOpenFileSymlinks/directFileOpenAllowed
    --- FAIL: TestOpenFileSymlinks/directFileOpenAllowed (0.00s)
```

See the suggestion above. For this particular test, I think this works:

```suggestion
	t.Run("directFileOpenAllowed", func(t *testing.T) {
		filePath, err = filepath.EvalSymlinks(filePath)
		require.NoError(t, err)
		f, err := openFile(filePath, false)
		require.NoError(t, err)
		defer f.Close()
	})
```

Alternatively, you may want to apply this right next to declaration of `filePath` variable:

```golang
	filePath := filepath.Join(dirPath, "file")
	_, err = os.Create(filePath)
	require.NoError(t, err)
	filePath, err = filepath.EvalSymlinks(filePath)
	require.NoError(t, err)
```